modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/AndesKernelBoot.java

```diff
@@ -237,70 +253,100 @@ private static MessageStore createMessageStoreFromConfig(AndesContextStore andes
         Andes.getInstance().initialise(subscriptionEngine, messagingEngine,
                 contextInformationManager, subscriptionManager);
 
         // initialize amqp constructs syncing into Qpid
-        VirtualHostConfigSynchronizer _VirtualHostConfigSynchronizer = new
-                VirtualHostConfigSynchronizer(
-                virtualHost);
-        ClusterResourceHolder.getInstance()
-                             .setVirtualHostConfigSynchronizer(_VirtualHostConfigSynchronizer);
+        VirtualHostConfigSynchronizer _VirtualHostConfigSynchronizer = new VirtualHostConfigSynchronizer(virtualHost);
+        ClusterResourceHolder.getInstance().setVirtualHostConfigSynchronizer(_VirtualHostConfigSynchronizer);
+    }
+
+    /**
+     * Start all andes stores message store/context store and AMQP construct store
+     *
+     * @throws Exception
+     */
+    public static void startAndesStores() throws Exception {
+
+        //Create a andes context store and register
+        AndesContextStore contextStoreInConfig = createAndesContextStoreFromConfig();
+        AndesKernelBoot.contextStore = new FailureObservingAndesContextStore(contextStoreInConfig);
+        AndesContext.getInstance().setAndesContextStore(contextStore);
+
+        // directly wire the instance without wrapped instance
+        messageStore = new FailureObservingMessageStore(createMessageStoreFromConfig(contextStoreInConfig));
+        // Setting the message store in the context store
+        AndesContext.getInstance().setMessageStore(messageStore);
+
+        //create AMQP Constructs store
+        AMQPConstructStore amqpConstructStore = new AMQPConstructStore(contextStore, messageStore);
+        AndesContext.getInstance().setAMQPConstructStore(amqpConstructStore);
     }
 
+    /**
+     * Initialize mode of cluster event synchronization depending on configurations and start listeners.
+     */
+    private static void initClusterEventSynchronization() throws AndesException {
+
+        if (ClusterResourceHolder.getInstance().getClusterManager().isClusteringEnabled()) {
+            ClusterEventSyncMode clusterEventSyncMode = AndesConfigurationManager.readValue
+                    (AndesConfiguration.CLUSTER_EVENT_SYNC_MODE);
+            if (ClusterEventSyncMode.RDBMS.equals(clusterEventSyncMode)) {
+                eventListenerCreator = new RDBMSBasedEventListenerCreator();
+                log.info("Broker is initialized with RDBMS based cluster event synchronization.");
+                (new RDBMSClusterNotificationListener()).initializeListener();
+
+            } else {
+                eventListenerCreator = new HazelcastBasedEventListenerCreator();
+                log.info("Broker is initialized with HAZELCAST based cluster event synchronization.");
+                HazelcastAgent.getInstance().initializeListener();
+            }
+        } else {
+            eventListenerCreator = new StandaloneEventListenerCreator();
+        }
```

Comment:
No we don't have to do that. We can create the listener creater in the cluster manager?